docs/guides/dependency-installation.rst

```diff
@@ -0,0 +1,223 @@
+
+
+``PythonImports``
+-----------------
+
+As the name suggests, the ``PythonImports`` importer handles ingestion of python modules, specifically from PyPi.
+
+The required modules can be specified as arguments to the importer, or listed in a requirements.txt file specified by the ``file`` argument.
+
+The examples below use ``PythonImports`` as the source importer for consistency with the other examples in this section. In actual usage, however, python imports will almost always be a dependency rather than a source.
+
+.. code-block:: python
+    import orquestra.sdk as sdk
+
+    # Individually specified as arguments
+    @sdk.task(
+        source_import=sdk.PythonImports(
+            "typing_extensions==4.5.0", "pytest~=6.2.5", "gpustat>=1.0.0"
+        )
+    )
+    def demo_task_1():
+        pass
+
+    # Read from file
+    @sdk.task(source_import=sdk.PythonImports(file = "./requirements.txt"))
+    def demo_task_2():
+        pass
+
+    # Both
+    @sdk.task(
+        source_import=sdk.PythonImports(
+            "typing_extensions==4.5.0", "pytest~=6.2.5", "gpustat>=1.0.0",
+            file = "./requirements.txt",
+        )
+    )
+    def demo_task_1():
+        pass
+
+The specified modules will be pip installed at execution time.
+
+``InlineImport``
+----------------
+
+Cases where the ``InlineImport`` importer must be specified explicitly are rare, as the SDK will set it as the default source importer in contexts where it applies. We include it here for completeness.
+
+The ``InlineImport`` importer ingests sources or dependencies that are defined in the ``__main__`` of the file that defines the workflow. This highly specific use case occurs almost exclusively in Jupyter Notebooks and interactive python sessions, however it is occasionally useful in rapid development or prototyping contexts. In general, the SDK's help messages will flag up instances where it should be used.
+
+The local importer
+------------------
+
+The local importer is the default source importer where no other importer is specified. It imports the current repo as a module, making its methods available to tasks in local workflow runs.
+
+
+Good Practice
+-------------
+
+For workflows running remotely the ``GitHubImport`` importer should be used unless there is a specific reason to do otherwise (i.e. the code is hosted somewhere other than GitHub). ``GitImport.infer()`` is highly user-specific and so should never be used in a collaborative context.
+
+For workflows running locally, the task decorator can typically be used without specifying the source importer as the SDK will set this appropriately for the context:
+
+.. code-block:: python
+    @sdk.task
+    def my_task():
+        pass
+
+will perform a local import unless running in a Jupyter Notebook (or other interactive session) in which case am inline import will be performed.
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**thought:** ‏Should this be first?

Reply:
I wAlmost certainly, especially with the quick reference tl;dr up top. Fixed in [2337332](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/74/commits/2337332f42a2571eead93827cf1d58ae61ec6a0e)